src/tribler/gui/start_gui.py

```diff
@@ -38,12 +38,14 @@ def run_gui(api_port: Optional[int], api_key: Optional[str], root_state_dir, par
         logger.info('Enabling a workaround for Ubuntu 21.04+ wayland environment')
         os.environ["GDK_BACKEND"] = "x11"
 
-    current_process = TriblerProcess.current_process(ProcessKind.GUI)
-    process_manager = ProcessManager(root_state_dir, current_process)
+    process_lock = try_acquire_file_lock(root_state_dir / GUI_LOCK_FILENAME)
+    current_process_has_lock = bool(process_lock)
+
+    process_manager = ProcessManager(root_state_dir)
+    process_manager.setup_current_process(kind=ProcessKind.GUI, has_lock=current_process_has_lock)
     set_global_process_manager(process_manager)  # to be able to add information about exception to the process info
-    current_process_is_primary = process_manager.current_process.become_primary()
 
-    load_logger_config('tribler-gui', root_state_dir, current_process_is_primary)
+    load_logger_config('tribler-gui', root_state_dir, current_process_has_lock)
```

Comment:
The name `current_process_has_lock` seems ambiguous (the same name also used on the core side). 
It's unclear whether the current process has detected an existing lock (indicating that it is a secondary) or if it is acquiring the lock (indicating it is primary).

Possible alternatives:
* `current_process_is_primary`
* `lock_has_been_acquired`
* `current_process_owns_lock`

Reply:
Agreed, renamed it to `current_process_owns_lock`